### PR TITLE
Add extra properties to doc generation

### LINKED ISSUE #29: Fill in all properties on document generation
See the following;
```
# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
# could be handy for archiving the generated documentation or if some version
# control system is used.

PROJECT_NUMBER         =

# Using the PROJECT_BRIEF tag one can provide an optional one line description
# for a project that appears at the top of each page and should give viewer a
# quick idea about the purpose of the project. Keep the description short.

PROJECT_BRIEF          =

# With the PROJECT_LOGO tag one can specify a logo or an icon that is included
# in the documentation. The maximum height of the logo should not exceed 55
# pixels and the maximum width should not exceed 200 pixels. Doxygen will copy
# the logo to the output directory.

PROJECT_LOGO           =
```

### PR DESCRIPTION
**Pull request by paddatrapper**
Fixes #29

Still need to add the project number, as I am unsure what number to use

### CODE REVIEW

**Comment by paddatrapper** [comment]
@mithro This now includes the SHA1 hash as the project number and is ready for review/merge

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,8 @@ $(FX2LIBDIR)/.git: .gitmodules
 
 # FIXME: Add check_int2jit from hdmi2usb/Makefile
 
+docs: export PROJECT_NUMBER:=$(shell git rev-parse HEAD; git diff-index --quiet HEAD || echo "(with uncommitted changes)")
```
Comment (mithro) [inline Makefile]: You probably want to replace `git rev-parse HEAD; git diff-index --quiet HEAD || echo "(with uncommitted changes)` with `git describe --always --dirty --long`

**Review by mithro — COMMENTED** [review]
Just a single comment, then I will merge.

File: Makefile (hunk 1 above)
Comment (paddatrapper) [reply Makefile]: Fixed

**Comment by paddatrapper** [comment]
Fixed

**Review by mithro — APPROVED** [review]
Lgtm

**Pull request merged**